app/abci/vote_extension.go

```diff
@@ -0,0 +1,74 @@
+package abci
+
+import (
+	cometabci "github.com/cometbft/cometbft/abci/types"
+
+	"cosmossdk.io/log"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/sedaprotocol/seda-chain/app/utils"
+)
+
+type VoteExtensionHandler struct {
+	batchingKeeper BatchingKeeper
+	pubKeyKeeper   PubKeyKeeper
+	signer         utils.SEDASigner
+	logger         log.Logger
+}
+
+func NewVoteExtensionHandler(bk BatchingKeeper, pkk PubKeyKeeper, signer utils.SEDASigner, logger log.Logger) *VoteExtensionHandler {
+	return &VoteExtensionHandler{
+		batchingKeeper: bk,
+		pubKeyKeeper:   pkk,
+		signer:         signer,
+		logger:         logger,
+	}
+}
+
+func (h *VoteExtensionHandler) ExtendVoteHandler() sdk.ExtendVoteHandler {
+	return func(ctx sdk.Context, _ *cometabci.RequestExtendVote) (*cometabci.ResponseExtendVote, error) {
+		h.logger.Debug("start extend vote handler")
+		batch, err := h.batchingKeeper.GetCurrentBatch(ctx)
+		if err != nil {
+			return nil, err
+		}
+		if batch.BlockHeight != ctx.BlockHeight()-1 {
+			return nil, ErrNoBatchForCurrentHeight
+		}
```

Comment:
I think it's a little clearer if we make it explicit that we want the batch of the previous height. So something like `batchingKeeper.GetBatchForHeight(ctx, ctx.BlockHeight()-1)`.

I'm not sold on that approach btw, I just want to avoid confusion where we say current, but then verify if it's the currentHeight-1. I'm also fine with something like `GetPreviousBlockBatch` on the Keeper and have that take care of the height shenanigans.

Reply:
Perhaps it's the error message that should be changed? In our design, a batch will be created at every block, and validators will always be signing a batch created from the previous height, right?

Reply:
Now the batches are primarily indexed by the block height and additionally indexed by the reference key batch number.